pulsar-io/kafka-connect-adaptor/src/main/java/org/apache/pulsar/io/kafka/connect/KafkaConnectSink.java

```diff
@@ -290,4 +299,25 @@ protected long currentOffset(String topic, int partition) {
         return taskContext.currentOffset(topic, partition);
     }
 
+    // Replace all non-letter, non-digit characters with underscore.
+    // Append underscore in front of name if it does not begin with alphabet or underscore.
+    protected String sanitizeNameIfNeeded(String name, boolean sanitize) {
```

Comment:
What about making this 'static' and add unit tests about edge cases? Empty name, null name, use of cache, whole name without good characters....

Reply:
static - doesn't matter. It uses cache instance, so it is either a static that accepts ache as a parameter or not static.

unit tests - if the topic name is blank or null we already have problem somewhere. Variations of characters - look at the regex, it is either a-z/0-9 or not, no need to test for "." vs "/" vs "$" (or whatever else it's allowed in the topic name)